src/setup_payload/tests/TestAdditionalDataPayload.cpp

```diff
@@ -0,0 +1,197 @@
+    uint16_t lifetimeCounter = 10;
+    BitFlags<AdditionalDataFields> additionalDataFields;
+
+    // Init Chip memory management before the stack
+    chip::Platform::MemoryInit();
+
+    err = AdditionalDataPayloadGenerator().generateAdditionalDataPayload(lifetimeCounter, serialNumber, ArraySize(serialNumber),
+                                                                         bufferHandle, additionalDataFields);
+    NL_TEST_ASSERT(inSuite, err == CHIP_NO_ERROR);
+    NL_TEST_ASSERT(inSuite, !bufferHandle.IsNull());
+
+    char output[51];
+    err = BytesToUppercaseHexString(bufferHandle->Start(), bufferHandle->DataLength(), output, ArraySize(output));
+
+    NL_TEST_ASSERT(inSuite, err == CHIP_NO_ERROR);
+    NL_TEST_ASSERT(inSuite, strcmp(output, "55000018") == 0);
+}
+
+void TestGeneratingAdditionalDataPayloadWithRotatingIdAndParsingIt(nlTestSuite * inSuite, void * inContext)
+{
+    CHIP_ERROR err = CHIP_NO_ERROR;
+    chip::System::PacketBufferHandle bufferHandle;
+
+    char serialNumber[]      = "89051AAZZ236";
+    uint16_t lifetimeCounter = 10;
+    BitFlags<AdditionalDataFields> additionalDataFields;
+    additionalDataFields.Set(AdditionalDataFields::RotatingDeviceId);
+
+    // Init Chip memory management before the stack
+    chip::Platform::MemoryInit();
+
+    err = AdditionalDataPayloadGenerator().generateAdditionalDataPayload(lifetimeCounter, serialNumber, ArraySize(serialNumber),
+                                                                         bufferHandle, additionalDataFields);
+    NL_TEST_ASSERT(inSuite, err == CHIP_NO_ERROR);
+    NL_TEST_ASSERT(inSuite, !bufferHandle.IsNull());
+
+    char output[51];
+    err = BytesToUppercaseHexString(bufferHandle->Start(), bufferHandle->DataLength(), output, ArraySize(output));
+
+    NL_TEST_ASSERT(inSuite, err == CHIP_NO_ERROR);
+    NL_TEST_ASSERT(inSuite, strcmp(output, "5500003000120A00D1414E39C9A08469DA2E29F0ACAA086618") == 0);
+
+    AdditionalDataPayload resultPayload;
+    err = AdditionalDataPayloadParser(bufferHandle->Start(), bufferHandle->DataLength()).populatePayload(resultPayload);
+    NL_TEST_ASSERT(inSuite, err == CHIP_NO_ERROR);
+    NL_TEST_ASSERT(inSuite, strcmp(resultPayload.rotatingDeviceId.c_str(), "0A00D1414E39C9A08469DA2E29F0ACAA0866") == 0);
+}
+
+void TestGeneratingAdditionalDataPayloadWithoutRotatingIdAndParsingIt(nlTestSuite * inSuite, void * inContext)
+{
+    CHIP_ERROR err = CHIP_NO_ERROR;
+    chip::System::PacketBufferHandle bufferHandle;
+
+    char serialNumber[]      = "89051AAZZ236";
+    uint16_t lifetimeCounter = 10;
+    BitFlags<AdditionalDataFields> additionalDataFields;
+
+    // Init Chip memory management before the stack
+    chip::Platform::MemoryInit();
+
+    err = AdditionalDataPayloadGenerator().generateAdditionalDataPayload(lifetimeCounter, serialNumber, ArraySize(serialNumber),
+                                                                         bufferHandle, additionalDataFields);
```

Comment:
This API is really odd: you pass in data but it is fully ignored. Suggest making the testcase take-in nullptr serial number and 0 lifetime counter and length of serial number, or better yet, remove the concept of the flags, and use a new `.addRotatingId` field to set the data appropriately...

Reply:
updated the test cases for now, i can take changing the API in a separate PR